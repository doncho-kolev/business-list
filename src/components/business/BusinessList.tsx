import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Row } from '../../features/list/Row';
import ListComponent from '../../features/list/ListComponent';
import { Business } from '../../models/Business';

export default function BusinessList() {
    const businesses: Business[] = [];
    const navigate = useNavigate();
    return (
        <div>
            <Row cells={[
                { label: "Name" },
                { label: "Description" }
            ]}
                className={"header-row"}
            />
            <ListComponent rows={businesses.map(b => {
                return {
                    cells: [
                        { label: b.name },
                        { label: b.description || "" },
                    ],
                    onClick: () => {
                        navigate(`/business/${b.id}`)
                    }
                }
            })} />
        </div>
    );
}
