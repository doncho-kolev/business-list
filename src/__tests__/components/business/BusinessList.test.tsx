import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import BusinessList from '../../../components/business/BusinessList';
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn()
}))
import * as Router from 'react-router-dom';

describe("Businesses List", () => {

    test('renders businesses list and click on entry', async () => {
        const valueGetterSpy = jest.fn();
        valueGetterSpy.mockReturnValue("Test Return Value");
        const routerSpy = jest.fn();
        jest.spyOn(Router, 'useNavigate').mockImplementation(() => routerSpy);
        const mount = render(<BusinessList businesses={[
            {
                id: "testId",
                name: "Test Name"
            }
        ]}
            valueGetter={valueGetterSpy}
        />);
        expect(mount.asFragment().children[0]).toMatchSnapshot();
        const row = await mount.findByText("Test Name");
        await userEvent.click(row);
        expect(routerSpy).toHaveBeenCalledWith(`/business/testId`);
        
    });
});
