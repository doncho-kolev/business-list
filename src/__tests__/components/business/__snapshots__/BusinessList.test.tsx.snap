// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Businesses List renders businesses list and click on entry 1`] = `
<div>
  <div>
    <div
      class=" row"
    >
      <div
        class="name-cell cell"
      >
        Test Name
      </div>
      <div
        class="description-cell cell"
      >
        Test Return Value
      </div>
    </div>
  </div>
</div>
`;
